pkg/kubelet/kuberuntime/labels.go

```diff
@@ -105,6 +106,14 @@ func newContainerLabels(container *v1.Container, pod *v1.Pod, containerType kube
 	labels[types.KubernetesContainerNameLabel] = container.Name
 	if utilfeature.DefaultFeatureGate.Enabled(features.DebugContainers) {
 		labels[types.KubernetesContainerTypeLabel] = string(containerType)
+		// It might be useful to store the spec for every container, but for now only store spec for debug containers
+		if containerType == kubecontainer.ContainerTypeEphemeral {
+			if containerJSON, e := json.Marshal(container); e != nil {
+				glog.Errorf("Unable to marshal %v container %q of pod %v: %v", containerType, container, pod.Name, e)
```

Comment:
nit: use [`format.Pod(pod)`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/util/format/pod.go#L32) in place of the pod name.